plugin/CMakeLists.txt

```diff
@@ -6,3 +6,16 @@ endif (PLUGIN_LZ4)
 if (PLUGIN_DENSE_PARSER)
   target_sources(objxgboost PRIVATE ${xgboost_SOURCE_DIR}/plugin/dense_parser/dense_libsvm.cc)
 endif (PLUGIN_DENSE_PARSER)
+
+if (PLUGIN_UPDATER_ONEAPI)
+  
+  set(PLUGINS_SOURCES_ONEAPI 
+    ${xgboost_SOURCE_DIR}/plugin/updater_oneapi/regression_obj_oneapi.cc 
+    ${xgboost_SOURCE_DIR}/plugin/updater_oneapi/predictor_oneapi.cc)
+
+  set(PLUGINS_SOURCES_ONEAPI ${PLUGINS_SOURCES_ONEAPI} PARENT_SCOPE)
```

Comment:
@vepifanov  Let's make a separate object target `oneapi_plugin`, so that we can call `set_source_files_properties()` on this target. Then we don't have to move `objxgboost`. If you'd like, I can make the change.

Reply:
That would be nice, thank you!
I thought about making another target, but I'm not sure how to properly transfer necessary flags from base target.

Reply:
@vepifanov I just pulled [the OneAPI Docker image](https://hub.docker.com/r/intel/oneapi-basekit) from Intel. I am getting this error when I invoked CMake:
```
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.18/Modules/FindOpenMP.cmake:529 (find_package_handle_standard_args)
  CMakeLists.txt:114 (find_package)


-- Configuring incomplete, errors occurred!
```

How did you get OpenMP working with Intel compilers?

Reply:
@vepifanov I found the fix:
```
cmake .. -DPLUGIN_UPDATER_ONEAPI=ON -DCMAKE_C_COMPILER=clang \
    -DCMAKE_CXX_COMPILER=dpcpp -DOpenMP_CXX_LIB_NAMES="libiomp5"  \
    -DOpenMP_C_LIB_NAMES="libiomp5" -DOpenMP_CXX_FLAGS="-openmp"  \
    -DOpenMP_C_FLAGS="-openmp"   \
    -DOpenMP_libiomp5_LIBRARY=/opt/intel/inteloneapi/compiler/latest/linux/compiler/lib/intel64_lin/libiomp5.so
```

Reply:
Good thing that Intel provides a Docker image. We might be able to add oneAPI plugin to CI in the near future.

Reply:
See my latest commit.